### DISCUSSION TITLE
Per Tenant Authentication

### DISCUSSION THREAD

**Discussion by technical101** [Finbuckle/Finbuckle.MultiTenant#596] [Q&A]
Hi,

I'm currently using ASP.NET identity for authentication for a multi tenant app.  Each customer has their own database and that works fine using Finbuckle.

My question is has any one managed to successfully integrate per tenant authentication with OpenIdConnect but also using ASP.NET identity.  What I'm after is having ASP.NET identity as the default option but with the ability to show "Login with AAD" for example based on the tenant.  What is currently happening is the tenant will automatically load the OpenIdConnect and redirect rather than enable the user to log in both ways.

I should also add that I'm using .NET 6, the "Host" strategy and "EFStore".

Thanks

**Comment by AndrewTriesToCode** [comment]
Hi, you should be able to do this by registering opened connect as usual, using `WithPerTenantAuthentication`, and `WithPerTenantOptions` for OpenID Connect options. That way you can set different OpenID Connect options for each tenant -- which authority they should use, client id, and so on. You'd need to store these somewhere, either in your tenant info implementation or some other configuration or database where you know what each tenant's opened connect options should be.

Check the .NET Core 3.1 samples, there is a project that should be helpful as a guide.
